src/main/java/com/couchbase/client/commons/CouchbaseArrayList.java

```diff
@@ -97,4 +111,106 @@ public Object remove(int index) {
             }
         }
     }
+
+    @Override
+    public boolean contains(Object o) {
+        //TODO in Spock subdoc may have ARRAY_CONTAINS which can help implement indexOf, remove(Object) and contains
+        return super.contains(o);
+    }
+
+    @Override
+    public Iterator<E> iterator() {
+        return new CouchbaseListIterator(0);
+    }
+
+    @Override
+    public ListIterator<E> listIterator() {
+        return new CouchbaseListIterator(0);
+    }
+
+    @Override
+    public ListIterator<E> listIterator(int index) {
+        return new CouchbaseListIterator(index);
+    }
+
+    @Override
+    public void clear() {
+        lastCas = bucket.upsert(JsonArrayDocument.create(id, JsonArray.empty())).cas();
+    }
+
+    private class CouchbaseListIterator implements ListIterator<E> {
+
+        private final long cas;
+        private final ListIterator<E> delegate;
+        int lastVisited;
+
+        public CouchbaseListIterator(int index) {
+            JsonArrayDocument current = bucket.get(id, JsonArrayDocument.class);
+            CouchbaseArrayList.this.lastCas = current.cas();
+            this.cas = current.cas();
+            this.delegate = ((List<E>) current.content().toList()).listIterator(index);
+            this.lastVisited = -1;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return delegate.hasNext();
+        }
+
+        @Override
+        public E next() {
+            lastVisited = nextIndex();
+            return delegate.next();
+        }
+
+        @Override
+        public boolean hasPrevious() {
+            return delegate.hasPrevious();
+        }
+
+        @Override
+        public E previous() {
+            lastVisited = delegate.previousIndex();
+            return delegate.previous();
+        }
+
+        @Override
+        public int nextIndex() {
+            return delegate.nextIndex();
+        }
+
+        @Override
+        public int previousIndex() {
+            return delegate.previousIndex();
+        }
+
+        @Override
+        public void remove() {
+            if (lastVisited < 0) {
+                throw new IllegalStateException();
+            }
+            int index = lastVisited;
+            lastVisited = -1;
+            String idx = "[" + index + "]";
+            try {
+                bucket.mutateIn(id).remove(idx).withCas(this.cas).execute();
+            } catch (CASMismatchException ex) {
+                throw new ConcurrentModificationException();
+            } catch (MultiMutationException ex) {
+                if (ex.firstFailureStatus() == ResponseStatus.SUBDOC_PATH_NOT_FOUND) {
+                    throw new ConcurrentModificationException("Element doesn't exist anymore at index: " + index);
+                }
+            }
+        }
+
+        @Override
+        public void set(E e) {
+            throw new UnsupportedOperationException();
```

Comment:
for now unsupported, until I think more about it 😄 maybe it will turn out we can't easily support it though...